flow/Platform.actor.cpp

```diff
@@ -3725,12 +3725,11 @@ void registerCrashHandler() {
 	sigaction(SIGSEGV, &action, nullptr);
 	sigaction(SIGBUS, &action, nullptr);
 	sigaction(SIGUSR2, &action, nullptr);
-#ifdef USE_GCOV
-	// SIGTERM is the "graceful" way to end an fdbserver process, so we actually
-	// don't want to invoke crashHandler. crashHandler is not actually
-	// async-signal-safe, so we can only justify calling it if we were going to
-	// crash anyway. For USE_GCOV though we need to flush coverage info, which
-	// we do through crashHandler.
+// SIGTERM was disabled except for USE_GCOV case to avoid a TSAN error when fdbmonitor kills a server.
+// It is, however, also causing servers to die without flushing their trace, loss of which harms auditing and
+// diagnostics. As a workaround, we specifically disable SIGTERM handler for sanitizer builds
+// TODO: fix TSAN error associated with kill by SIGTERM
+#if !defined(USE_SANITIZER) || defined(USE_GCOV)
 	sigaction(SIGTERM, &action, nullptr);
```

Comment:
I don't think it is safe to revert this behavior. SIGTERM was causing sporadic deadlocks, because the signal handler in fdbserver does non-trivial stuff that are not signal-handler safe. I don't see why this issue is specific to TSAN builds, it could be that it is more likely to happen in TSAN builds.

Reply:
As discussed offline, let's work on this in a separate PR.